pkg/cmd/diff/diff.go

```diff
@@ -0,0 +1,159 @@
+package diff
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/MaineK00n/vuls-data-update/pkg/cmd/util/flag"
+	"github.com/MaineK00n/vuls-data-update/pkg/diff"
+	"github.com/MaineK00n/vuls-data-update/pkg/extract/util"
+	"github.com/MakeNowJust/heredoc"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+)
+
+type option struct {
+	org       string
+	dir       string
+	filter    string
+	diffAlg   string
+	format    flag.Choices
+	outputDir string
+}
+
+func NewCmdDiff() *cobra.Command {
+	opts := &option{
+		org:     "vulsio",
+		dir:     filepath.Join(util.CacheDir(), "diff"),
+		filter:  "tree:0",
+		diffAlg: "default",
```

Comment:
Is it necessary to impose restrictions on the values ​​that can be entered, such as format?

Reply:
Hmm... it may be or may not, depending on cases...
These two options are not used in the abstraction layer of this code itself, but just pass them to underlying layer, git command. I'm thinking of the case where git adds new options.  In that case, strict restrictions blocks users to use new options until code is fixed.
My feeling is: it's good to help users for shell completions, but strict restriction is not very useful.

I noticed, at this point, I failed to add completion func for filter option, oops.

Reply:
For those who want to use the new `--filter` or `--diff-algorithm` and are reasonably familiar with git, why don't they just send us a Pull Request?
I'd rather not have people who aren't familiar with that sort of thing have trouble with it. (I don't think those people would specify the option)

Reply:
done in https://github.com/MaineK00n/vuls-data-update/compare/0337bc95d8cb49c0ed17e0a3f0b36a39c7456436..eaa71234f1324c709d312d3c0b9a38824bc47649